src/components/__tests__/useResource.test.tsx

```diff
@@ -0,0 +1,43 @@
+import { Mock } from 'firebase-nightlight';
+import { renderHook } from '@testing-library/react-hooks';
+import { foodResource as mockResource } from '../../DataMocks';
+
+// see https://github.com/kulshekhar/ts-jest/issues/120#issuecomment-283653644
+// for some helpful info about some typescript nonsense with jest
+
+jest.mock('../../firebase', () => {
+  mockApp = new Mock(); // eslint-disable-line @typescript-eslint/no-use-before-define
+  return {
+    __esModule: true,
+    default: mockApp.initializeApp({}) // eslint-disable-line @typescript-eslint/no-use-before-define
+  };
+});
+
+import useResource from '../useResource';
+
+var mockApp: Mock; // using `var` here so it is hoisted above the mock -- would love a better solution
```

Comment:
Unfortunately, I don't have a better solution for ensuring that variables are defined before declaring a mock. Something I have seen kind of work before is [doMock](https://jestjs.io/docs/en/jest-object.html#jestdomockmodulename-factory-options), which allows you to dynamically update mocks. You just have to reimport the module being tested.

So it would be something like:

```js
const mockApp = new Mock();

jest.doMock('../whateverModule', () => ({
  __esModule: true,
  default: mockApp.initializeApp({})
}));

const useResource = require('../useResource'); // using require since import is hoisted (I think)

//kewl tests here
```

This does mean you that you have to use `require` as well which also goes against our current conventions 🙁

Reply:
Yeah this is all pretty shabby..